src/core/node/index.ts

```diff
@@ -535,7 +535,12 @@ export class ElementNode extends Object {
   render() {
     // eslint-disable-next-line @typescript-eslint/no-this-alias
     const node = this;
-    const parent = this.parent!;
+    const parent = this.parent;
+
+    // prevents Too many redirects error
+    if (!parent) {
```

Comment:
you should never have this error unless you're creating nodes without a parent and you should be researching why you're doing that.